File: 159/made_by_JS/String_Palindrome.js
const func_main = (source) => {
	const str = source.replace(/\r?\n/g,"");
	if (!(check_str(str, str.split("").reverse().join("")))) {
		console.log("No");
	} else {
		console.log("Yes");
	}
}

const check_str = (str1, str2) => {
	if (str1 == str2) {
		return true;
	} else {
		return false;
	}
}

func_main(require('fs').readFileSync('/dev/stdin', 'utf8'));

File: 159/made_by_Python/String_Palindrome.py
def judge_str(str1, str2):
	print("str1 :{},str2 :{}".format(str1, str2))
	if str1 == str2:
		return 1
	else:
		return 0

S = list(map(str, input()))
print(S)
N = len(S)
print(N)
if not judge_str(S[:], S[::-1]):
	print("No")
else:
	print("Yes")
